js/bybit.js

```diff
@@ -3670,7 +3693,182 @@ module.exports = class bybit extends Exchange {
+        //    }
+        //
+        const contract = this.safeString (position, 'symbol');
+        market = this.safeMarket (contract, market);
+        const size = this.safeString (position, 'size');
+        let side = this.safeString (position, 'side');
+        side = (side === 'Buy') ? 'long' : 'short';
+        const notional = this.safeString2 (position, 'position_value', 'positionValue');
+        const unrealisedPnl = this.safeString2 (position, 'unrealised_pnl', 'unrealisedPnl');
+        const initialMarginString = this.safeString2 (position, 'position_margin', 'orderMargin');
+        const percentage = Precise.stringMul (Precise.stringDiv (unrealisedPnl, initialMarginString), '100');
+        let timestamp = this.parse8601 (this.safeString (position, 'updated_at'));
+        if (timestamp === undefined) {
+            timestamp = this.safeInteger (position, 'createdAt');
+            if (timestamp === undefined) {
+                timestamp = this.milliseconds ();
+            }
+        }
+        const isIsolated = this.safeValue (position, 'is_isolated', false); // if not present it is cross
+        const marginMode = isIsolated ? 'isolated' : 'cross';
+        return {
+            'info': position,
+            'symbol': this.safeString (market, 'symbol'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'initialMargin': this.parseNumber (initialMarginString),
+            'initialMarginPercentage': this.parseNumber (Precise.stringDiv (initialMarginString, notional)),
+            'maintenanceMargin': undefined,
+            'maintenanceMarginPercentage': undefined,
+            'entryPrice': this.safeNumber2 (position, 'entry_price', 'entryPrice'),
+            'notional': this.parseNumber (notional),
+            'leverage': this.safeNumber (position, 'leverage'),
+            'unrealizedPnl': this.parseNumber (unrealisedPnl),
+            'contracts': this.parseNumber (size), // in USD for inverse swaps
+            'contractSize': this.safeNumber (market, 'contractSize'),
+            'marginRatio': undefined,
+            'liquidationPrice': this.safeNumber2 (position, 'liq_price', 'liqPrice'),
+            'markPrice': this.safeNumber (position, 'markPrice'),
+            'collateral': undefined,
+            'marginMode': marginMode,
+            'side': side,
+            'percentage': this.parseNumber (percentage),
+        };
+    }
+
+    parsePositions (positions, market = undefined) {
+        const result = [];
+        for (let i = 0; i < positions.length; i++) {
+            let rawPosition = positions[i];
+            if (('data' in rawPosition) && ('is_valid' in rawPosition)) {
+                // futures only
+                rawPosition = this.safeValue (rawPosition, 'data');
+            }
+            const size = this.safeNumber (rawPosition, 'size');
+            // Bybit returns all positions possibilities so we have to
```

Comment:
Let's not do this filtering – we do not decide whether we should return just the open positions or all of them, it is up to the exchange.